components/scream/src/share/io/scream_scorpio_interface.F90

```diff
@@ -689,6 +693,7 @@ subroutine eam_init_pio_subsystem(mpicom,atm_id,local)
       pio_subsystem  => shr_pio_getiosys(atm_id)
       pio_iotype     = shr_pio_getiotype(atm_id)
       pio_rearranger = shr_pio_getrearranger(atm_id)
+      pio_mode       = shr_pio_getioformat(atm_id)
     else
 #else
       ! Just for removing unused dummy warnings
```

Comment:
If I understand correctly, this `#else` in the `#ifdef CIME_BUILD` doesn't do anything at all and the `if (.false.)` line is just so there's *something* in the else statement so the compiler doesn't complain. One obvious question is why not just delete the `#else` section entirely. The more relevant question for this PR is why not set `pio_mode = PIO_64BIT_DATA` in that `#else` statement rather than overwriting all of the `#ifdef` pio_mode logic above by simply hardcoding pio_mode on line 707?

Reply:
I can see how this set of lines is confusing.  But the way I read it.  If CIME_BUILD is true then we end up having a `if(.not.local)` conditional.  Which allows us to choose between grabbing the pio info from the `shr` modules, or in the case of the `else` statement, doing it ourselves.

If CIME_BUILD is false then we do have the `if(.false.)` statement, but it looks like that is just to remove compiler warnings.  The main part is that in CIME_BUILD false we default to making the PIO system ourselves, which is needed for stand_alone tests.  So perhaps a better question would be, what is the purpose of passing `local`, in what case would we have a CIME_BUILD with `local=.true.`?

Reply:
whoa, this is spaghetti code. I missed the `if/else` within the `#if/#else` block! Yeah, I agree that the `local` flag seems redundant with CIME_BUILD.

I looked at the git blame for this file and it looks like you put in the local flag and Luca added the CIME_BUILD stuff.

Should we just integrate the PR as-is and create a low-priority issue of detangling this section of code?